Include jackson-dataformat-xml.

From: Lars Francke <git@lars-francke.de>

This is needed for XmlLayout to work so our structured logging works.
It is an optional dependency of log4j2 so we need to make sure to include
it.
---
 hbase-assembly/pom.xml |    5 +++++
 pom.xml                |   12 ++++++++++++
 2 files changed, 17 insertions(+)

diff --git a/hbase-assembly/pom.xml b/hbase-assembly/pom.xml
index 1564851b85..0786288a84 100644
--- a/hbase-assembly/pom.xml
+++ b/hbase-assembly/pom.xml
@@ -222,6 +222,11 @@
       <groupId>org.apache.logging.log4j</groupId>
       <artifactId>log4j-core</artifactId>
     </dependency>
+    <dependency>
+      <!-- This is an optional dependency of log4j which is needed to use XmlLayout -->
+      <groupId>com.fasterxml.jackson.dataformat</groupId>
+      <artifactId>jackson-dataformat-xml</artifactId>
+    </dependency>
     <dependency>
       <groupId>org.apache.logging.log4j</groupId>
       <artifactId>log4j-slf4j-impl</artifactId>
diff --git a/pom.xml b/pom.xml
index b420025c6c..819e021d86 100644
--- a/pom.xml
+++ b/pom.xml
@@ -1149,6 +1149,11 @@
         <artifactId>log4j-core</artifactId>
         <version>${log4j2.version}</version>
       </dependency>
+      <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-core</artifactId>
+        <version>${log4j2.version}</version>
+      </dependency>
       <dependency>
         <groupId>org.apache.logging.log4j</groupId>
         <artifactId>log4j-slf4j-impl</artifactId>
@@ -1159,6 +1164,13 @@
         <artifactId>log4j-1.2-api</artifactId>
         <version>${log4j2.version}</version>
       </dependency>
+      <dependency>
+        <!-- This is an optional dependency of log4j which is needed to use XmlLayout -->
+        <groupId>com.fasterxml.jackson.dataformat</groupId>
+        <artifactId>jackson-dataformat-xml</artifactId>
+        <version>${jackson.databind.version}</version>
+      </dependency>
+
       <!-- Avro dependencies we mostly get transitively, manual version coallescing -->
       <dependency>
         <groupId>org.apache.avro</groupId>
